lib/spack/spack/util/environment.py

```diff
@@ -25,23 +25,10 @@
 import os
 
 system_paths = ['/', '/usr', '/usr/local']
-suffixes = ['lib', 'lib64', 'include']
-system_dirs = [os.path.join(p, s) for s in suffixes for p in system_paths] + \
-    system_paths
-system_bins = [os.path.join(p, 'bin') for p in system_paths]
 
 
 def filter_system_paths(paths):
-    return [p for p in paths if p not in system_dirs]
```

Comment:
my point is -- you don't need to change the function to make the PR do the job? So it does filter extra things (if they are there), but we currently don't use it this way. I would argue to keep the extra (unused) functionality.